internal/compiler/utils.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2023 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package compiler
+
+import (
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/util"
+)
+
+func VerifyAuthorizationPolicySchema(rules []*ast.Rule) error {
+
+	if len(rules) == 0 {
+		return nil
+	}
+
+	inputSchema := `{
+	 "$schema": "http://json-schema.org/draft-04/schema#",
+	 "type": "object",
+	 "properties": {
+	   "identity": {
+	     "type": "string"
+	   },
+	   "client_certificates": {
+	     "type": "array",
+	     "items": {}
+	   },
+	   "method": {
+	     "type": "string"
+	   },
+	   "path": {
+	     "type": "array",
+	     "items": {}
+	   },
+	   "params": {
+	     "type": "object"
+	   },
+	   "headers": {
+	     "type": "object"
+	   },
+	   "body": {
+	     "type": "object"
+	   }
+	 },
+	 "required": [
+	   "identity",
+	   "client_certificates",
+	   "method",
+	   "path",
+	   "params",
+	   "headers",
+	   "body"
+	 ]
+	}`
+
+	var schema interface{}
+	if err := util.Unmarshal([]byte(inputSchema), &schema); err != nil {
```

Comment:
Since the schema is static, perhaps we should avoid unmarshalling in each call and instead do that once beforehand?

Reply:
Yes. Updated.